vmdb/app/models/mixins/ems_openstack_mixin.rb

```diff
@@ -27,7 +27,9 @@ def openstack_handle(options = {})
       raise "no credentials defined" if self.missing_credentials?(options[:auth_type])
 
       username = options[:user] || self.authentication_userid(options[:auth_type])
-      password = options[:pass] || self.authentication_password(options[:auth_type])
+
+      # the |"" is here to prevent an unhelpful ArgumentError from OpenStackHandle::Handle#connect
+      password = options[:pass] || self.authentication_password(options[:auth_type]) || ""
```

Comment:
Although @blomquisg I think @Fryguy @roliveri did something with amazon with authentication classes for kinds that don't need userid.  I think this is in the same area of what we're saying here.  

Either way, I think the provider should say what type of authentication they need (requires userid and password, only userid, some other value) and authentications should support those types...

I think?  I still don't know what to do with code that did .blank? since password is blank for both "" and nil.

what do you think?